examples/ssl/train_hubert.py

```diff
@@ -0,0 +1,304 @@
+from torchaudio.models import (
+    hubert_pretrain_base,
+    hubert_pretrain_large,
+    hubert_pretrain_xlarge,
+)
+
+
+class _Formatter(ArgumentDefaultsHelpFormatter, RawDescriptionHelpFormatter):
+    # To use ArgumentDefaultsHelpFormatter as the formatter_class and
+    # RawDescriptionHelpFormatter to add custom formatting to description or epilog.
+    # Check: https://stackoverflow.com/a/18462760
+    pass
+
+
+def _compute_accuracy(logits: torch.Tensor):
+    with torch.no_grad():
+        max = logits.argmax(-1) == 0
+        min = logits.argmin(-1) == 0
+        both = max & min
+        corr = max.long().sum().item() - both.long().sum().item()
+        count = max.numel()
+    return corr / count
+
+
+class HuBERTModule(SSLPretrainModule):
+    def configure_optimizers(self):
+        return (
+            [self.optimizer],
+            [
+                {
+                    "scheduler": self.lr_scheduler,
+                    "interval": "step",
+                },
+            ],
+        )
+
+    def log_metric(
+        self, batch: Dict, output: Tuple, loss: torch.Tensor, step_type: str
+    ):
+        logit_m, logit_u, _ = output
+        self.log(
+            f"{step_type}_loss",
+            loss.item(),
+            on_step=True,
+            on_epoch=True,
+        )
+        acc_m = _compute_accuracy(logit_m)
+        acc_u = _compute_accuracy(logit_u)
+        self.log(
+            f"{step_type}_acc_m",
+            acc_m,
+            on_step=True,
+            on_epoch=True,
+            sync_dist=True,
+            prog_bar=step_type == "train",
+        )
+        self.log(
+            f"{step_type}_acc_u",
+            acc_u,
+            on_step=True,
+            on_epoch=True,
+            sync_dist=True,
+            prog_bar=step_type == "train",
+        )
+
+    def get_sample_size(self, output):
+        logit_m, _, _ = output
+        return logit_m.shape[0]
+
+
+def run_train(args):
+    seed_everything(1337)
+    checkpoint_dir = args.exp_dir / f"checkpoints_{args.dataset}_{args.model_name}"
+    checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="val_loss",
+        mode="min",
+        save_top_k=5,
+        save_weights_only=False,
+        verbose=True,
+    )
+    train_checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="train_loss",
+        mode="min",
+        save_top_k=5,
+        save_weights_only=False,
+        verbose=True,
+    )
+    callbacks = [
+        checkpoint,
+        train_checkpoint,
+    ]
+    trainer = Trainer(
+        default_root_dir=args.exp_dir,
+        max_steps=args.max_updates,
+        num_nodes=args.num_nodes,
+        devices=args.gpus,
+        accelerator="gpu",
+        strategy="ddp",
+        replace_sampler_ddp=False,
+        callbacks=callbacks,
+        reload_dataloaders_every_n_epochs=1,
+    )
+    if args.model_name == "hubert_pretrain_base":
```

Comment:
can do assert args.model_name in [...] and then use getattr instead of multiple if statements

Reply:
cool, will change in that way